tests/python/pants_test/engine/test_scheduler.py

```diff
@@ -61,6 +84,16 @@ def test_use_params(self):
     with self.assertRaises(Exception):
       self.scheduler.product_request(str, [Params(a)])
 
+  def test_transitive_params(self):
+    a, c = A(), C()
+    result_str, = self.scheduler.product_request(str, [Params(a, c)])
+    self.assertEquals(remove_locations_from_traceback(result_str),
+                      remove_locations_from_traceback(consumes_a_and_b(a, transitive_b_c(c))))
+
+    result_d, = self.scheduler.product_request(D, [Params(c)])
+    self.assertEquals(remove_locations_from_traceback(str(result_d)),
+                      remove_locations_from_traceback(str(D())))
```

Comment:
I don't think the `remove_locations_from_traceback` calls here are necessary since we're comparing the `str` representation of what should be two `D()`s

Reply:
Yes -- the idea was to use it to remove the memory location info. I think as you say we can just ensure that they are both `D()`s.